File: david/Question.java
import java.util.ArrayList;

public class Question {
  
  public String question, answer;
  public ArrayList<String> options = new ArrayList<>();

  public Question(String question){
    this.question = question;
  }
  public void add_option(String option){
    options.add(option);
  }
  public void set_answer(String answer){
    this.answer = answer;
  }
  public void print_question(){
    System.out.println(question);
    char opt = 'a';
    for(String s : options){
      System.out.println("" + opt + ". " + s);
      opt++;
    }
    System.out.println("");
  }
  public String wrap(){ // join question and all options into one string
    String res = question + "|";
    for(String s : options) {
      res += s;
      res += "|";
    }
    return res;
  }
  public static Question unWrap(String msg) { 
    Question res = new Question("");

    int last = 0;
    for(int i = 0; i < msg.length(); i++){
      if(msg.charAt(i) == '|'){
        // question part
        if(last == 0){
          res.question = msg.substring(0, i);
          last = i + 1;
        }
        // answer part
        else {
          res.add_option(msg.substring(last, i));
          last = i + 1;
        }
      }
    }
    return res;
  }

}

File: david-python/question.py
class Question:
  def __init__(self, question):
    self.question = question
    self.options = []
    self.answer = ""
  
  def add_option(self, opt):
    self.options.append(opt)
  
  def print_question(self):
    print("Question: " + self.question)
    opsi = 97
    for o in self.options:
      print(chr(opsi) + ". " + o)
      opsi += 1 
    print("")

  def wrap(self):
    res = self.question
    for o in self.options:
      res += "|" + o
    res += "|"
    return res
  
  def unwrap(qq):
    last = 0
    ret = Question("")
    for i in range(len(qq)):
      if qq[i] != '|': continue
      # question
      if last == 0:
        ret.question = qq[0:i]
        last = i + 1
      # option
      else :
        ret.add_option(qq[last:i])
        last = i + 1
    return ret

# q = Question("Test")
# q.add_option("david ganten")
# q.add_option("bisa ggak")
# q.print_question()  

# tmp = Question.unwrap(q.wrap())
# tmp.print_question()


